internal/service/ivschat/logging_configuration.go

```diff
@@ -0,0 +1,426 @@
+	conn := meta.(*conns.AWSClient).IVSChatClient
+
+	update := false
+
+	in := &ivschat.UpdateLoggingConfigurationInput{
+		Identifier: aws.String(d.Id()),
+	}
+
+	if d.HasChanges("name") {
+		in.Name = aws.String(d.Get("name").(string))
+		update = true
+	}
+
+	if d.HasChanges("destination_configuration") {
+		in.DestinationConfiguration = expandDestinationConfiguration(d.Get("destination_configuration").([]interface{}))
+		update = true
+	}
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+
+		if err := UpdateTags(ctx, conn, d.Id(), o, n); err != nil {
+			return create.DiagError(names.IVS, create.ErrActionUpdating, ResNameLoggingConfiguration, d.Id(), err)
+		}
+	}
+
+	if !update {
+		return nil
+	}
+
+	log.Printf("[DEBUG] Updating IVSChat LoggingConfiguration (%s): %#v", d.Id(), in)
+	out, err := conn.UpdateLoggingConfiguration(ctx, in)
+	if err != nil {
+		return create.DiagError(names.IVSChat, create.ErrActionUpdating, ResNameLoggingConfiguration, d.Id(), err)
+	}
+
+	if _, err := waitLoggingConfigurationUpdated(ctx, conn, aws.ToString(out.Arn), d.Timeout(schema.TimeoutUpdate)); err != nil {
+		return create.DiagError(names.IVSChat, create.ErrActionWaitingForUpdate, ResNameLoggingConfiguration, d.Id(), err)
+	}
+
+	return resourceLoggingConfigurationRead(ctx, d, meta)
+}
+
+func resourceLoggingConfigurationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).IVSChatClient
+
+	log.Printf("[INFO] Deleting IVSChat LoggingConfiguration %s", d.Id())
+
+	_, err := conn.DeleteLoggingConfiguration(ctx, &ivschat.DeleteLoggingConfigurationInput{
+		Identifier: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		var nfe *types.ResourceNotFoundException
+		if errors.As(err, &nfe) {
+			return nil
+		}
+
+		return create.DiagError(names.IVSChat, create.ErrActionDeleting, ResNameLoggingConfiguration, d.Id(), err)
+	}
+
+	if _, err := waitLoggingConfigurationDeleted(ctx, conn, d.Id(), d.Timeout(schema.TimeoutDelete)); err != nil {
+		return create.DiagError(names.IVSChat, create.ErrActionWaitingForDeletion, ResNameLoggingConfiguration, d.Id(), err)
+	}
+
+	return nil
+}
+
+func flattenDestinationConfiguration(apiObject types.DestinationConfiguration) []interface{} {
+	if apiObject == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{}
+
+	switch v := apiObject.(type) {
+	case *types.DestinationConfigurationMemberCloudWatchLogs:
+		m["cloudwatch_logs"] = flattenCloudWatchDestinationConfiguration(v.Value)
+
+	case *types.DestinationConfigurationMemberFirehose:
+		m["firehose"] = flattenFirehoseDestinationConfiguration(v.Value)
+
+	case *types.DestinationConfigurationMemberS3:
+		m["s3"] = flattenS3DestinationConfiguration(v.Value)
+
+	case *types.UnknownUnionMember:
+		log.Println("unknown tag:", v.Tag)
+
+	default:
+		log.Println("union is nil or unknown type")
+	}
+
+	return []interface{}{m}
+}
```

Comment:
This pattern was taken from https://github.com/hashicorp/terraform-provider-aws/blob/d1e68d3f2987a8e75401d39df7b226ebb3d7c5b2/internal/service/rolesanywhere/trust_anchor.go#L231-L250